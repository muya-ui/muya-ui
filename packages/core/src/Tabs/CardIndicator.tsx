import styled, { css } from 'styled-components';

import { IThemedBaseProps } from '../types';
import { withThemeForStyled } from '../utils/withTheme';
import { ITabsProps } from './types';

function indicatorCss(props: ITabsProps & IThemedBaseProps) {
  const { theme } = props;
  const { Tabs: token } = theme.components;
  const { indicator } = token;
  return css`
    position: absolute;
    width: 100%;
    bottom: 0;
    left: 0;
    height: ${indicator.bgHeight}px;
    background: ${indicator.bgColor};
  `;
}

const CardIndicator = styled.div`
  ${indicatorCss}
`;
export default withThemeForStyled(CardIndicator);
