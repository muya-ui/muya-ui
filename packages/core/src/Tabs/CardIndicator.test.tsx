import React from 'react';
import renderer from 'react-test-renderer';

import CardIndicator from './CardIndicator';

test('测试 CardIndicator 默认', () => {
  const tree = renderer.create(<CardIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});
