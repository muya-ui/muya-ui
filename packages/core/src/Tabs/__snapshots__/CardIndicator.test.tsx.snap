// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 CardIndicator 默认 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background: rgba(238,238,238,1);
}

<div
  className="c0"
/>
`;
